packages/flutter/test/painting/text_painter_test.dart

```diff
@@ -952,6 +952,22 @@ void main() {
       expect(glyphBox, newGlyphBox);
     });
   }, skip: isBrowser);
+
+  test('TextPainter handles invalid UTF-16', () {
+    Object? exception;
+    FlutterError.onError = (FlutterErrorDetails details) {
+      exception = details.exception;
+    };
+
+    final TextPainter painter = TextPainter()
+      ..textDirection = TextDirection.ltr;
+
+    String text = 'Hello\uD83DWorld';
+    painter.text = TextSpan(text: text);
+    painter.layout();
+    expect(painter.width, greaterThan(0.0));
```

Comment:
it should be exactly 10.0, right? assuming default font size is 10 and font is Ahem

Reply:
done